interfaces/builtin/dpdk_observe.go

```diff
@@ -0,0 +1,49 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const dpdkObserveSummary = `allows using dpdk`
+
+const dpdkObserveBaseDeclarationSlots = `
+  dpdk-observe:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const dpdkObserveConnectedPlugAppArmor = `
+# Description: Allow access to dpdk.
+/sys/kernel/mm/hugepages/{,**} r,
+/dev/hugepages/{,**} r,
+/run/dpdk/{,**} r,
+/proc/{,**} r,
```

Comment:
Can you provide the rationale for reading _everything_ in proc? This feels too powerful to me.

Reply:
During my testing, I was running into issues where dpdk could not read it's proc running and would cause a core dump. I was hoping that there was already an interface in place for this (process-control or system-observe) but I had to explicitly set it in the dpdk interface for it to function properly.